util/progress/progress.go

```diff
@@ -180,7 +189,14 @@ func pipe() (*progressReader, *progressWriter, func()) {
 	return pr, pw, cancel
 }
 
-func newWriter(pw *progressWriter) *progressWriter {
+// NewWriter returns a new progress writer based on the Writer argument.
+// It is the callers responsibility to call Close on the Writer to prevent
+// resource leaks.
+func NewWriter(w Writer, opts ...WriterOption) Writer {
```

Comment:
Generally ok with this approach but this public API makes little sense. Both `New` and `From` return writer, `NewWriter` takes writer as a parameter. I think better would be if `FromContext` returns `ProgressFactory`, a function that returns a new writer when called. `NewWriter` is not public.

Reply:
Okay, updated.  I went with `WriterFactory` to avoid the stutter with `progress.ProgressFactory`.